pkg/component/registrycaches/registry_caches.go

```diff
@@ -380,6 +380,27 @@ func (r *registryCaches) computeResourcesDataForRegistryCache(ctx context.Contex
 		},
 	}
 
+	if cache.Proxy != nil {
+		if cache.Proxy.HTTPProxy != nil {
+			statefulSet.Spec.Template.Spec.Containers[0].Env = append(statefulSet.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+				Name:  "HTTP_PROXY",
+				Value: *cache.Proxy.HTTPProxy,
+			})
+		}
+		if cache.Proxy.HTTPSProxy != nil {
+			statefulSet.Spec.Template.Spec.Containers[0].Env = append(statefulSet.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+				Name:  "HTTPS_PROXY",
+				Value: *cache.Proxy.HTTPSProxy,
+			})
+		}
+		if cache.Proxy.NoProxy != nil {
+			statefulSet.Spec.Template.Spec.Containers[0].Env = append(statefulSet.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+				Name:  "NO_PROXY",
+				Value: *cache.Proxy.NoProxy,
+			})
+		}
```

Comment:
According to the implementation only one of the `.httpProxy`, `.httpsProxy`, `.noProxy` fields can be specified. You could add a validation about it in https://github.com/gardener/gardener-extension-registry-cache/blob/1300d37d60e4578db3eb807e3bd119c7997e9340/pkg/apis/registry/validation/validation.go#L25.

Reply:
No it is possible to add all 3 variables to the container, as it is using `append()` to add the ENV to the slice. Or do I get something wrong?